In [1]:
%load_ext watermark
%watermark

2019-05-22T06:46:41+02:00

CPython 3.6.8
IPython 6.2.1

compiler   : GCC 7.3.0
system     : Linux
release    : 4.13.0-46-generic
machine    : x86_64
processor  : x86_64
CPU cores  : 2
interpreter: 64bit


## Carga Dataframe

In [2]:
import pandas as pd
import datetime
import warnings

warnings.filterwarnings('ignore')

In [3]:
df = pd.read_pickle('../Data/01.-Exploracion-de-los-Datos-y-Creacion-Dataframe.pkl')

df.head()

Fecha_envio Product_code Origen Destino  Peso  Volumen  Peso_final
0  2017-05-01            B     ES      NO  1.80    3.584       3.584
1  2017-05-01            B     ES      NO  0.90    2.242       2.242
2  2017-05-01            B     ES      NO  0.35    1.009       1.009
3  2017-05-01            B     ES      NO  1.15    2.130       2.130
4  2017-05-01            B     ES      NO  0.45    0.996       0.996

## Encontrar producto estrella por país 

In [4]:
df['Product_code'].unique()

array(['B', 'U', 'P', 'D', 'N', 'C', 'T', 'W', 'H', 'Q', '7', '9', '8',
       '1', 'X', 'K', 'V', 'Y', 'I', 'L', 'M', 'E'], dtype=object)

In [5]:
df_shippings = df.groupby(['Destino','Product_code']).size().reset_index()

df_shippings.head()

Destino Product_code     0
0      AD            7   192
1      AD            8    16
2      AD            D  7177
3      AD            H     3
4      AD            N     4

In [6]:
df_shippings.columns = ['Destino','Producto','Total']

df_shippings = df_shippings.sort_values(by = 'Total',ascending = False)

df_shippings.head()

Destino Producto    Total
422      ES        N  1163861
470      FR        U  1157595
705      IT        U   974413
345      DE        U   771564
495      GB        U   717266

In [7]:
df_shippings_ES = df_shippings[df_shippings['Destino'] == 'ES']

df_shippings_ES.head()

Destino Producto    Total
422      ES        N  1163861
412      ES        1   117037
415      ES        C    45819
413      ES        7    41009
416      ES        D     4775

In [8]:
df_shippings_ES['Porcentaje_del_total'] = ((df_shippings_ES['Total'] / df_shippings_ES['Total'].sum()) * 100).round(2)

df_shippings_ES.head()

Destino Producto    Total  Porcentaje_del_total
422      ES        N  1163861                 84.40
412      ES        1   117037                  8.49
415      ES        C    45819                  3.32
413      ES        7    41009                  2.97
416      ES        D     4775                  0.35

El 'Producto' N es del más envíos se realiza, teniendo en cuenta la segmentación de países. Y el país, en este caso, es España, para el que supone el 84.40% de los envíos.

Vamos a crear un DataFrame únicamente para los campos 'Fecha_envio' y 'Total', para el 'Producto' N en el 'Destino' ES.

## Creación DataFrame 'Producto' N con 'Destino' ES 

In [9]:
df.head()

Fecha_envio Product_code Origen Destino  Peso  Volumen  Peso_final
0  2017-05-01            B     ES      NO  1.80    3.584       3.584
1  2017-05-01            B     ES      NO  0.90    2.242       2.242
2  2017-05-01            B     ES      NO  0.35    1.009       1.009
3  2017-05-01            B     ES      NO  1.15    2.130       2.130
4  2017-05-01            B     ES      NO  0.45    0.996       0.996

In [10]:
df_N_ES = df[(df['Product_code'] == 'N') & (df['Destino'] == 'ES')]

df_N_ES.head()

Fecha_envio Product_code Origen Destino  Peso  Volumen  Peso_final
1234  2017-05-02            N     ES      ES  1.60    0.500       1.600
1279  2017-05-02            N     ES      ES  5.78    1.075       5.780
1281  2017-05-02            N     ES      ES  0.70    1.575       1.575
1282  2017-05-02            N     ES      ES  0.96    0.500       0.960
1286  2017-05-02            N     ES      ES  0.38    0.500       0.500

In [12]:
df_N_ES = df_N_ES.groupby(['Fecha_envio']).size().reset_index()

df_N_ES.head()

Fecha_envio     0
0  2017-05-02  1993
1  2017-05-03  3193
2  2017-05-04  2759
3  2017-05-05  2503
4  2017-05-06    56

In [14]:
df_N_ES.columns = ['ds','y']

df_N_ES.head()

ds     y
0 2017-05-02  1993
1 2017-05-03  3193
2 2017-05-04  2759
3 2017-05-05  2503
4 2017-05-06    56

## Creación de Archivo con Formato pickle del Dataframe 

In [18]:
df_N_ES.to_pickle('../Data/02.-Untitled.pkl')

In [19]:
!ls -l --block-size=M ../Data/02.-Untitled.pkl

-rw-r--r-- 1 dsc dsc 1M may 22 07:10 ../Data/02.-Untitled.pkl
